bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BObjectType.java

```diff
@@ -136,8 +137,30 @@ public void setTypeIdSet(BTypeIdSet typeIdSet) {
         this.typeIdSet = typeIdSet;
     }
 
-    public void processObjectCtorAnnots(MapValue globalAnnotationMap, Strand strand) {
-        AnnotationUtils.processObjectCtorAnnotations(globalAnnotationMap, this, strand);
+    public BObjectType duplicateTypeAndprocessObjectCtorAnnots(MapValue globalAnnotationMap, Strand strand) {
```

Comment:
```suggestion
    public BObjectType duplicateTypeAndProcessObjectCtorAnnots(MapValue globalAnnotationMap, Strand strand) {
```

Reply:
Does this method belong here? Wouldn't it be better to have this in `AnnotationUtils`?

Reply:
Please elaborate why you think this doesn't belong here.

Reply:
IMO, even though annotations are part of the type, this duplication and reprocessing the annotations is an approach we're adopting to address a particular scenario.

I believe the `BObjectType` does not have to know about processing annotations, instead we can address this via `AnnotationUtils` which handles all of the annotation processing and setting it to the types - i.e., `AnnotationUtils` calls `duplicate()` on the `BObjectType` and sets the annotations to the new type. The only exception to this seems to be `io.ballerina.runtime.internal.types.BObjectType#processObjectCtorAnnots` which I think could also happen via `AnnotationUtils`.

I think the eventual approach to this should be to codegen the creation of a new `BObjectType` for each new (service) object constructor, and let the annotation processing happen as for a class/normal object via `AnnotationUtils`.

Reply:
Refactored this and code-gen so to directly call duplicate() and then process annot on that duplicated type. Please check.

IIRC I moved this method to BObjectType from BServiceType class since we need to do this duplication stuff for object constructor as well. Agreed with your reasoning.